src/versions/drafts/buildDraftMergeAggregate.ts

```diff
@@ -0,0 +1,79 @@
+import { PipelineStage } from 'mongoose';
+import { SanitizedCollectionConfig } from '../../collections/config/types';
+
+type Args = {
+  config: SanitizedCollectionConfig
+  query: Record<string, unknown>
+}
+
+export const buildDraftMergeAggregate = ({ config, query }: Args): PipelineStage[] => [
+  // Merge in versions by parent ID
+  {
+    $lookup: {
+      from: `_${config.slug}_versions`,
+      as: '_versions',
+      localField: '_id',
+      foreignField: 'parent',
+    },
+  },
+
+  // Add a new field
+  // with the last (newest) version returned
+  {
+    $addFields: {
+      _version: {
+        $arrayElemAt: ['$_versions', -1],
```

Comment:
is _versions always sorted by updatedAt/createdAt

If it sorts by created at will there be problems if a "versions" can be updated later on?